internal/pkg/deploy/cloudformation/stack/scheduled_job.go

```diff
@@ -0,0 +1,315 @@
+	cronWeekly  = "0 0 ? * 1 *" // at midnight on sunday
+	cronMonthly = "0 0 1 * ? *" // at midnight on the first of the month
+	cronYearly  = "0 0 1 1 ? *" // at midnight on January 1
+)
+
+const (
+	hourly   = "@hourly"
+	daily    = "@daily"
+	midnight = "@midnight"
+	weekly   = "@weekly"
+	monthly  = "@monthly"
+	yearly   = "@yearly"
+	annually = "@annually"
+
+	every = "@every "
+)
+
+// NewScheduledJob creates a new ScheduledJob stack from a manifest file.
+func NewScheduledJob(mft *manifest.ScheduledJob, env, app string, rc RuntimeConfig) (*ScheduledJob, error) {
+	parser := template.New()
+	addons, err := addon.New(aws.StringValue(mft.Name))
+	if err != nil {
+		return nil, fmt.Errorf("new addons: %w", err)
+	}
+	envManifest, err := mft.ApplyEnv(env)
+	if err != nil {
+		return nil, fmt.Errorf("apply environment %s override: %w", env, err)
+	}
+	return &ScheduledJob{
+		wkld: &wkld{
+			name:   aws.StringValue(mft.Name),
+			env:    env,
+			app:    app,
+			tc:     envManifest.ScheduledJobConfig.TaskConfig,
+			rc:     rc,
+			parser: parser,
+			addons: addons,
+		},
+		manifest: envManifest,
+
+		parser: parser,
+	}, nil
+}
+
+// Template returns the CloudFormation template for the scheduled job.
+func (j *ScheduledJob) Template() (string, error) {
+	outputs, err := j.addonsOutputs()
+	if err != nil {
+		return "", err
+	}
+
+	sidecars, err := j.manifest.Sidecar.Options()
+	if err != nil {
+		return "", fmt.Errorf("convert the sidecar configuration for job %s: %w", j.name, err)
+	}
+
+	schedule, err := j.awsSchedule()
+	if err != nil {
+		return "", fmt.Errorf("convert schedule for job %s: %w", j.name, err)
+	}
+
+	stateMachine, err := j.stateMachineOpts()
+	if err != nil {
+		return "", fmt.Errorf("convert retry/timeout config for job %s: %w", j.name, err)
+	}
+
+	content, err := j.parser.ParseScheduledJob(template.WorkloadOpts{
+		Variables:          j.manifest.Variables,
+		Secrets:            j.manifest.Secrets,
+		NestedStack:        outputs,
+		Sidecars:           sidecars,
+		ScheduleExpression: schedule,
+		StateMachine:       stateMachine,
+		LogConfig:          j.manifest.LogConfigOpts(),
+	})
+	if err != nil {
+		return "", fmt.Errorf("parse scheduled job template: %w", err)
+	}
+	return content.String(), nil
+}
+
+// awsSchedule converts the Schedule string to the format required by Cloudwatch Events
+// https://docs.aws.amazon.com/lambda/latest/dg/services-cloudwatchevents-expressions.html
+// Cron expressions must have an sixth "year" field, and must contain at least one ? (either-or)
+// in either day-of-month or day-of-week.
+// Day-of-week expressions are zero-indexed in Golang but one-indexed in AWS.
+// @every cron definition strings are converted to rates.
+// All others become cron expressions.
+func (j *ScheduledJob) awsSchedule() (string, error) {
+	if j.manifest.Schedule == "" {
+		return "", fmt.Errorf("missing required field \"schedule\" in manifest for job %s", j.name)
```

Comment:
doesn't matter for this but to make your life easier you can also do this:
```suggestion
		return "", fmt.Errorf(`missing required field "schedule" in manifest for job %s`, j.name)
```